src/python/pants/engine/process_test.py

```diff
@@ -411,3 +412,59 @@ def test_workspace_execution_support() -> None:
     assert result3.stderr.decode() == "this-goes-to-stderr\n"
     snapshot = rule_runner.request(Snapshot, [result3.output_digest])
     assert snapshot.files == ("capture-this-file",)
+
+
+@pytest.mark.parametrize(
+    "concurrency",
+    [
+        ProcessConcurrency.range(min=1),
+        ProcessConcurrency.range(max=2),
+        ProcessConcurrency.range(min=1, max=2),
+        ProcessConcurrency.exclusive(),
+        # Values larger than num cores still work (they get clamped to num cores)
+        ProcessConcurrency.range(min=10000),
+        ProcessConcurrency.range(max=10000),
+        ProcessConcurrency.range(min=100, max=200),
+    ],
+)
+def test_concurrency(rule_runner: RuleRunner, concurrency: ProcessConcurrency) -> None:
+    process = Process(
+        argv=("/bin/echo", "hello"),
+        concurrency=concurrency,
+        description="concurrency-test",
+    )
```

Comment:
So usage in a plugin would look like 

```py
process = Process(
    argv=("/bin/echo", "hello"),
    concurrency=ProcessConcurrency.range(min=4),
    description="concurrency-test",
)
```
or

```py
process = Process(
    argv=("/bin/echo", "hello"),
    concurrency=ProcessConcurrency.exclusive(),
    description="concurrency-test",
)
```

Which is a little wordy, but not too bad I guess

Reply:
Something like `ProcessConcurrency.range(min=2)` is reading awkwardly to me though. I don't think it conveys the meaningly clearly, that I need exactly 2 cpu cores. No more no less. Its not a range in that sense


Maybe something like `ProcessConcurrency.cores(2)` or `ProcessConcurrency.exactly(2)`

Reply:
Yeah, good point. I like `exactly` (since all the variants are about cores).